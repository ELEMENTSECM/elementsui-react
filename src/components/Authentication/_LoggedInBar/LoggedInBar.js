import * as React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'office-ui-fabric-react/lib/Link';
import { MessageBar, MessageBarType } from 'office-ui-fabric-react/lib/MessageBar';

/** LoggedInBar example */
function LoggedInBar({ name, handleLogoutClick, labels }) {
	return (
		<MessageBar messageBarType={MessageBarType.info}>
			{`${labels.loggedInAs} '${name}'. `}
			<Link href="" onClick={handleLogoutClick}>
				{labels.logout}
			</Link>
		</MessageBar>
	);
}

LoggedInBar.propTypes = {
	/** Logged in user's name */
	name: PropTypes.string,
	/** Log ou mouse click event handler */
	handleLogoutClick: PropTypes.func,
	/** Labels */
	labels: PropTypes.shape({
		/** Log out button label */
		logout: PropTypes.string,
		/** Logged in label */
		loggedInAs: PropTypes.string
	})
};

LoggedInBar.defaultProps = {
	labels: {
		logout: 'Log out',
		loggedInAs: 'You are logged in as'
	}
};

export default LoggedInBar;
