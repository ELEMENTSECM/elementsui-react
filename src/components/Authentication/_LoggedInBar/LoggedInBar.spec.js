import React from 'react';
import renderer from 'react-test-renderer';
import LoggedInBar from './LoggedInBar';

describe('LoggedInBar', () => {
	test('displays correctly', () => {
		const tree = renderer
			.create(
				<LoggedInBar name="John Doe" labels={{ logout: 'Sign out', loggedInAs: 'Logged in as' }} />
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
