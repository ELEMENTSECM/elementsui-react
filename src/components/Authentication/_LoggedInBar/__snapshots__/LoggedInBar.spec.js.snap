// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedInBar displays correctly 1`] = `
<div
  aria-live="polite"
  className="ms-MessageBar root_91b7bf6c ms-MessageBar root_91b7bf6c ms-MessageBar-multiline multiLine_91b7bf6c"
  role="status"
>
  <div
    className="content_91b7bf6c ms-MessageBar-content"
  >
    <div
      className="ms-MessageBar-icon icon_91b7bf6c"
    >
      <i
        aria-hidden={true}
        className="root-36"
        data-icon-name="Info"
        role="presentation"
      />
    </div>
    <div
      className="ms-MessageBar-text text_91b7bf6c noDismissButton_91b7bf6c"
      id="MessageBar0"
    >
      <span
        aria-live="polite"
        className="ms-MessageBar-innerText innerText_91b7bf6c"
        role="status"
      />
    </div>
  </div>
</div>
`;
